spec/controllers/basespace_controller_spec.rb

```diff
@@ -0,0 +1,239 @@
+        allow(ENV).to receive(:[]).with("BASESPACE_OAUTH_REDIRECT_URI").and_return("MOCK_URI")
+        allow(ENV).to receive(:[]).with("BASESPACE_CLIENT_ID").and_return("MOCK_ID")
+        allow(ENV).to receive(:[]).with("BASESPACE_CLIENT_SECRET").and_return("MOCK_SECRET")
+      end
+
+      it "renders the oauth template" do
+        get :oauth, params: { code: "MOCK_CODE" }
+        expect(response).to render_template("oauth")
+      end
+
+      it "assigns view variables correctly" do
+        get :oauth, params: { code: "MOCK_CODE" }
+        expect(assigns(:access_token)).to eq("12345")
+      end
+
+      context "when no code param is provided" do
+        it "assigns view variables correctly" do
+          get :oauth, params: {}
+          expect(response).to render_template("oauth")
+          expect(assigns(:access_token)).to eq(nil)
+        end
+      end
+    end
+
+    describe "GET projects" do
+      before do
+        allow(HttpHelper).to receive(:get).and_return("Response" => {
+                                                        "Items" => [
+                                                          {
+                                                            "Id" => "1",
+                                                            "Name" => "Project 1"
+                                                          },
+                                                          {
+                                                            "Id" => "2",
+                                                            "Name" => "Project 2"
+                                                          }
+                                                        ]
+                                                      })
+      end
+
+      it "returns projects" do
+        get :projects, params: { access_token: "123" }
+
+        json_response = JSON.parse(response.body)
+        expect(json_response).to include_json([
+                                                {
+                                                  "id" => "1",
+                                                  "name" => "Project 1"
+                                                },
+                                                {
+                                                  "id" => "2",
+                                                  "name" => "Project 2"
+                                                }
+                                              ])
+      end
+
+      context "when no access token is provided" do
+        it "returns an error" do
+          get :projects, params: {}
+
+          json_response = JSON.parse(response.body)
+          expect(json_response).to include_json(error: "basespace access token required")
+        end
+      end
+    end
+
+    describe "GET samples_for_project" do
+      before do
+        allow(HttpHelper).to receive(:get).and_return("Items" => [
```

Comment:
nit: Is there a way to reformat this to prevent rubocop from breaking this apart so much?